src/routes/safe/components/Transactions/TxsTable/ExpandedTx/index.jsx

```diff
@@ -61,10 +63,15 @@ const ExpandedTx = ({
   const classes = useStyles()
   const [openModal, setOpenModal] = useState<OpenModal>(null)
   const openApproveModal = () => setOpenModal('approveTx')
+  const openApproveCancelModal = () => setOpenModal('approveCancelTx')
   const openCancelModal = () => setOpenModal('cancelTx')
   const closeModal = () => setOpenModal(null)
   const thresholdReached = tx.type !== INCOMING_TX_TYPE && threshold <= tx.confirmations.size
   const canExecute = tx.type !== INCOMING_TX_TYPE && nonce === tx.nonce
+  const cancelThresholdReached = cancelTx
+    ? cancelTx.type !== INCOMING_TX_TYPE && threshold <= cancelTx.confirmations.size
+    : false
+  const canExecuteCancel = cancelTx ? cancelTx.type !== INCOMING_TX_TYPE && nonce === cancelTx.nonce : false
```

Comment:
why is there a possibility to get a cancelTx with type INCOMING_TX_TYPE? Couldn't this be filtered in a selector?

Reply:
This was me failing at review before commit it. If you give a look at it, it's just the same logic for the tx but replacing `tx` with `cancelTx` in several places.